### ISSUE TITLE
Compiler error resetting stringstream in src/framework/utils.hpp

### ISSUE THREAD

**Issue by jlatone** [Qiskit/qiskit-aer#851]
### Informations

- **Qiskit Aer version**: 0.5.2
- **Python version**: 3.7.6
- **Operating system**: Linux

### What is the current behavior?
The following compiler error from this line of code in **src/framework/utils.hpp:1499**  

> ss = std::stringstream(); // clear string stream

```
/u/jlatone/qiskitdeva/qiskit-aer/src/framework/utils.hpp:1499:25:   required from here
/u/jlatone/anaconda3a/envs/myqiskit/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/sstream:340:1: error: cannot call member function 'void std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::_M_pbump(std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::char_type*, std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::char_type*, std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::off_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::char_type = char; std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::off_type = long int]' without object
      _M_to->_M_pbump(__str+_M_poff[0], __str+_M_poff[2], _M_poff[1]);
 ^    ~~~

```

### Steps to reproduce the problem
Building/installing Aer exhibits the error, but it seems(?) to be an issue only when building Aer with GPU support since I haven't run into this in the past, i.e., 

> python ./setup.py -v install -- -DAER_THRUST_BACKEND=CUDA 

I tried several versions of g++, 4.4<=>8.4, and none of them support this method of resetting the object to its default ctor state, i.e., "clearing the string stream." A more compiler-portable way of doing it would be to either scope it inside the {}'s or ss.clear(); ss.str(""); (see below).

### What is the expected behavior?
No compiler error. 

### Suggested solutions
Scope it inside the {}'s or ss.clear(); ss.str(""); e.g., 

```
    for (size_t j=1; j < chunks; ++j) {
      std::stringstream ss; // clear string stream
      ss << std::hex << std::stoull(str.substr(remain + j * bin_block, bin_block), nullptr, 2);
      part = ss.str();
      part.insert(0, hex_block - part.size(), '0');
      hex += part;
    }
```
 
**-or-**

```
    for (size_t j=1; j < chunks; ++j) {
      ss.clear(); ss.str(""); // clear string stream
      ss << std::hex << std::stoull(str.substr(remain + j * bin_block, bin_block), nullptr, 2);
      part = ss.str();
      part.insert(0, hex_block - part.size(), '0');
      hex += part;
    }
```

**Comment by MarcusGDaniels** [comment]
I see before gcc 9.1 the assignment operator was absent.

*** gcc-8.2.0/libstdc++-v3/include/backward/strstream   2018-01-03 02:03:58.000000000 -0800
--- gcc-9.1.0/libstdc++-v3/include/backward/strstream   2019-03-21 05:07:10.000000000 -0700
***************
*** 167,172 ****
--- 216,229 ----
      strstream(char*, int, ios_base::openmode = ios_base::in | ios_base::out);
      virtual ~strstream();

+ #if __cplusplus >= 201103L
+     strstream(strstream&& __rhs)
+     : iostream(std::move(__rhs)), _M_buf(std::move(__rhs._M_buf))
+     { set_rdbuf(&_M_buf); }
+
+     strstream& operator=(strstream&&) = default;
+ #endif
+
      _GLIBCXX_CONST strstreambuf* rdbuf() const throw ();
      void freeze(bool = true) throw ();
      _GLIBCXX_PURE int pcount() const throw ();

**Comment by chriseclectic** [comment]
Fixed in #855
